GREYErrorFormatter.m

```diff
@@ -0,0 +1,120 @@
+//
+//  GREYErrorFormatter.m
+//  CommonLib
+//
+//  Created by Will Said on 6/1/20.
+//
+
+#import "GREYErrorFormatter.h"
+
+#import "GREYObjectFormatter.h"
+#import "GREYInteraction.h"
+#import "GREYFatalAsserts.h"
+#import "GREYErrorConstants.h"
+
+@interface GREYErrorFormatter ()
+
+@property(readonly, nonatomic) GREYError *error;
+
+@end
+
+@implementation GREYErrorFormatter
+
+#pragma mark - Class Methods
+
++ (NSString *)FormattedHierarchy:(nonnull NSString *)hierarchy {
+    NSMutableArray *logger = [[NSMutableArray alloc] init];
+    
+    [logger addObject:@"UI Hierarchy (ordered by window level, back to front):\n"];
+
+    NSString *windowLegend = @"[Window 1]";
+    NSString *axLegend = @"[AX]";
+    NSString *uieLegend = @"[UIE]";
+
+    NSDictionary *legendLabels = @{
```

Comment:
Moving forwards, when you're initializing containers, you should always specify generics.